src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java

```diff
@@ -0,0 +1,343 @@
+   * @param renameTo                the new file name, {@code null} if the file doesn't need to be renamed
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doCopy(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
+    validatePath(targetPath, "target path");
+    validatePath(sourcePath, "source path");
+    fileSystem.copy(config, sourcePath, targetPath, overwrite, createParentDirectories, renameTo);
+  }
+
+  /**
+   * Moves the file at the {@code sourcePath} into the {@code targetPath}.
+   * <p>
+   * If {@code targetPath} doesn't exist, and neither does its parent, then an attempt will be made to create depending on the
+   * value of the {@code createParentFolder} argument. If such argument is {@code false}, then an {@link IllegalArgumentException}
+   * will be thrown.
+   * <p>
+   * If the target file already exists, then it will be overwritten if the {@code overwrite} argument is {@code true}. Otherwise,
+   * {@link IllegalArgumentException} will be thrown.
+   * <p>
+   * As for the {@code sourcePath}, it can either be a file or a directory. If it points to a directory, then it will be moved
+   * recursively.
+   *
+   * @param config                  the config that is parameterizing this operation
+   * @param fileSystem              a reference to the host {@link FileSystem}
+   * @param sourcePath              the path to the file to be copied
+   * @param targetPath              the target directory
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param overwrite               whether or not overwrite the file if the target destination already exists.
+   * @param renameTo                the new file name, {@code null} if the file doesn't need to be renamed
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doMove(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
+    validatePath(targetPath, "target path");
+    validatePath(sourcePath, "source path");
+    fileSystem.move(config, sourcePath, targetPath, overwrite, createParentDirectories, renameTo);
+  }
+
+  /**
+   * Deletes the file pointed by {@code path}, provided that it's not locked
+   *
+   * @param fileSystem a reference to the host {@link FileSystem}
+   * @param path       the path to the file to be deleted
+   * @throws IllegalArgumentException if {@code filePath} doesn't exist or is locked
+   */
+  protected void doDelete(FileSystem fileSystem, @Optional String path) {
+    fileSystem.changeToBaseDir();
```

Comment:
again, absolute paths

Reply:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.